### DISCUSSION TITLE
Change language only server render

### DISCUSSION THREAD

**Discussion by jpsc** [nuxt-modules/i18n#2107] [General]
Hello,

I've been integrating this module (v8) into an application and I've noticed that there is (as expected) an added size to the initial entry bundle of a Nuxt 3 application. There are the translations and some runtime required features. Again that is totally normal.

While I was using my application I realized that we don't actually have the need to the language to change immediatly. It would be fine if we could have a hard page reset to load the new language. From my understanding this is how `vue-i18n` works. 

Does it make sense or would it be possible (from a concept perspective) to remove most of runtime and lower the amount of translations available because you already know you won't need them until you to a full page refresh?

Cheers

**Comment by zifnab87** [comment]
Hi jpsc no expert here but nuxt is a server side rendering tool .. and since i18n introduces new urls if there is javascript enabled then everything is loaded once and it acts like a Single Page application.. but if javascript is not enabled then all the translations should happen on the server side on each different url /en/page-1 /el/page-1 and work without javascript

Here are my findings so far (https://github.com/nuxt-modules/i18n/issues/2097#issuecomment-1562739152) and it seems i18n is not working as expected and it seems it also needs a server plugin to even load the translations when javascript is off.. this is important for all the bots that index your page.. the content and its translated versions need to preexist so your page can be indexed by search engines .. this is the main reason we use nuxt instead of vue since we need all those server side rendering capabilities for better SEO

**Comment by jpsc** [comment]
Hei @zifnab87 while I understand what you are saying the issue you reported is definitely a bug, what I'm talking is a bit different.
It should of course load the translations from the server-side (into initial html) but then I don't need the feature to change the language without a full page refresh. 

What I'm suggesting that without this requirement a lot of runtime code for `nuxt-i18n` (in reality vue-i18n) could be removed. But maybe I'm completly missing the mark.

**Reply by zifnab87** [reply]
ok I see @jpsc  I guess with a non default value for a flag, so you are requesting a feature.. most of us don't want a full page reload of the page just to change the locale though because it is really heavy and it essentially it behaves like SSR all the time when changing languages. I will wait to see what the module curators have to say

**Reply by jpsc** [reply]
@zifnab87, for sure i'm not imposing this feature for everybody 👍
I didn't put it on as a feature request because I was not sure if was already possible.

I'm thinking that once a user sets the language it will most likely will never change. He will though *pay the price* of always needing to parse code that he will not use.
